client-ts/signalr/src/JsonHubProtocol.ts

```diff
@@ -31,4 +58,29 @@ export class JsonHubProtocol implements IHubProtocol {
     public writeMessage(message: HubMessage): string {
         return TextMessageFormat.write(JSON.stringify(message));
     }
+
+    private isInvocationMessage(message: InvocationMessage): void {
+        if (message.target === undefined) {
+            throw new Error("Invalid payload for Invocation message.");
+        }
+    }
+
+    private isStreamItemMessage(message: StreamItemMessage): void {
+        if (message.invocationId === undefined) {
```

Comment:
Must be a `string`

Reply:
Maybe it is worth having a helper method.

```
private assertIsType(value: any, type: string, errorMessage: string) {
    if (typeof value !== type) {
        throw new Error(errorMessage);
    }
}

assertIsType(message.invocationId, "string", "Invalid payload for StreamItem message.");
assertIsType(message.item, "string", "Invalid payload for StreamItem message.");
```

Reply:
hmmm, but we also have to check the string has a value